src/ad-services/wunderkind/index.ts

```diff
@@ -0,0 +1,33 @@
+import { communicationService, eventsRepository, UapLoadStatus } from '@ad-engine/communication';
+import { BaseServiceSetup, context, utils } from '@ad-engine/core';
+
+const logGroup = 'wunderkind';
+
+export class Wunderkind extends BaseServiceSetup {
+	call(): void {
+		if (!this.instantConfig.get('icWunderkind')) {
+			utils.logger(logGroup, 'disabled');
+			return;
+		} else if (context.get('state.isLogged')) {
+			utils.logger(logGroup, 'disabled - user is logged');
+			return;
+		}
+
+		communicationService.on(eventsRepository.AD_ENGINE_UAP_LOAD_STATUS, (action: UapLoadStatus) => {
+			if (!action.isLoaded) {
+				this.loadScript();
+			} else {
+				utils.logger(logGroup, 'disabled - UAP is loaded');
+			}
```

Comment:
nitpick: quick return first instead of login inside if condition block? just for readability. 

```suggestion
			if (action.isLoaded) {
			    utils.logger(logGroup, 'disabled - UAP is loaded');
			    return;
			}
			this.loadScript();
```